src/steps/tmux.rs

```diff
@@ -144,25 +146,32 @@ pub fn run_in_tmux(args: Vec<String>) -> Result<()> {
         shell_words::join(command)
     };
 
-    let tmux = Tmux::new(args);
+    let tmux = Tmux::new(config.args);
 
     // Find an unused session and run `topgrade` in it with the current command's arguments.
     let session_name = "topgrade";
     let window_name = "topgrade";
     let session = tmux.new_unique_session(session_name, window_name, &command)?;
 
-    // Only attach to the newly-created session if we're not currently in a tmux session.
-    if env::var("TMUX").is_err() {
-        let err = tmux.build().args(["attach-session", "-t", &session]).exec();
-        Err(eyre!("{err}")).context("Failed to `execvp(3)` tmux")
-    } else {
-        println!("Topgrade launched in a new tmux session");
-        Ok(())
-    }
+    let is_inside_tmux = env::var("TMUX").is_ok();
+    let err = match config.session_attach_mode {
+        // Only attach to the newly-created session if we're not currently in a tmux session.
+        TmuxSessionAttachMode::Create if is_inside_tmux => {
+            println!("Topgrade launched in a new tmux session");
+            return Ok(());
+        }
+        TmuxSessionAttachMode::CreateAndSwitchClient if is_inside_tmux => {
+            tmux.build().args(["switch-client", "-t", &session]).exec()
+        }
+        TmuxSessionAttachMode::Create | TmuxSessionAttachMode::CreateAndSwitchClient => {
+            tmux.build().args(["attach-session", "-t", &session]).exec()
+        }
+    };
```

Comment:
What about writing it in this way, would it be a little bit clearer?

```rs
TmuxSessionAttachModule::Create => {
    if is_inside_tmux {
        println!("Topgrade launched in a new tmux session");
        return Ok(());
    } else {
        tmux.build().args(["attach-session", "-t", &session]).exec()
    }
}
TmuxSessionAttachModule::CreateAndSwitchClient => {
     if is_inside_tmux {
        tmux.build().args(["switch-client", "-t", &session]).exec()
     } else {
        tmux.build().args(["attach-session", "-t", &session]).exec()
     }
}
```

Reply:
I can do it if it's easier, just felt bad to repeat `attach-session` call twice

Reply:
> just felt bad to repeat attach-session call twice

Yeah, that's indeed one drawback. Though I prefer readability, which will make maintenance easier:)